src/librustc_typeck/check/demand.rs

```diff
@@ -58,7 +59,68 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
         if let Err(e) = self.try_coerce(expr, checked_ty, expected) {
             let origin = TypeOrigin::Misc(expr.span);
             let expr_ty = self.resolve_type_vars_with_obligations(checked_ty);
-            self.report_mismatched_types(origin, expected, expr_ty, e);
+            let suggestions = self.check_ref(expr, checked_ty, expected);
+            let mut err = self.report_mismatched_types(origin, expected, expr_ty, e);
+            if let Some(suggestions) = suggestions {
+                err.help(&suggestions);
```

Comment:
It's called `span_suggestion`: http://manishearth.github.io/rust-internals-docs/rustc/?search=fn%3A%20suggestion